File: Demo/Java/WSOTest.java
//WindowSystemObject (WSO) sample
//Copyright (C) Veretennikov A. B. 2004

import com.jacob.com.*;
import com.jacob.activeX.*;

public class WSOTest implements WSO.Constants
{

	static public void AboutWSO(Dispatch Sender, String Text)
	{
		Dispatch form = Dispatch.get(Sender,"Form").toDispatch();
		Dispatch j = Dispatch.call(form,"CreateDialogForm",new Variant(0),
		new Variant(0),new Variant(0),new Variant(0)).toDispatch();
		
		Dispatch.put(j,"ClientWidth",new Variant(400));
		Dispatch.put(j,"ClientHeight",new Variant(250));
		Dispatch.call(j,"CenterControl");
		
		Dispatch t = Dispatch.call(j,"Rectangle",new Variant(0),new Variant(0),
			new Variant(0),new Variant(75)).toDispatch();
		Dispatch b = Dispatch.get(t,"Brush").toDispatch();
		Dispatch.put(b,"Color",new Variant(0x00FFA0A0));
		Dispatch.put(b,"Style",new Variant(B_GRADIENT | DIRECTION_TOP_BOTTOM));
		Dispatch.put(b,"GradientColor",new Variant(0x00FFEEEE));
		Dispatch p = Dispatch.get(t,"Pen").toDispatch();
		Dispatch.put(p,"Color",new Variant(0x00FF5555));
		Dispatch.put(t,"Align",new Variant(AL_TOP));

		t = Dispatch.call(j,"Rectangle",new Variant(0),new Variant(0),
			new Variant(0),new Variant(0)).toDispatch();
		b = Dispatch.get(t,"Brush").toDispatch();
		Dispatch.put(b,"Color",new Variant(0x00FFEEEE));
		Dispatch.put(b,"Style",new Variant(B_GRADIENT | DIRECTION_TOP_BOTTOM));
		Dispatch.put(b,"GradientColor",new Variant(0x00FFEEEE));
		p = Dispatch.get(t,"Pen").toDispatch();
		Dispatch.put(p,"Color",new Variant(0x00FF5555));
		Dispatch.put(t,"Align",new Variant(AL_CLIENT));

		t = Dispatch.call(j,"CreateButton",new Variant(300),
		new Variant(220),new Variant(75),new Variant(25),new Variant("Close")).toDispatch();
		new DispatchEvents(t,new CloseItemEvents());
		Dispatch.put(t,"Default",new Variant(true));
		Dispatch.put(t,"Cancel",new Variant(true));		

		Dispatch Root = Dispatch.get(t,"Root").toDispatch();
		Dispatch Version = Dispatch.get(Root,"Version").toDispatch();
		String s = Dispatch.get(Version,"String").toString();
		
		t = Dispatch.call(j,"TextOut",new Variant(100),new Variant(10),new Variant(s)).toDispatch();
		Dispatch f = Dispatch.get(t,"Font").toDispatch();
		Dispatch.put(f,"Size",new Variant(14));
		Dispatch.put(f,"Bold",new Variant(true));
		Dispatch.put(f,"Name",new Variant("Arial Black"));
		Dispatch.put(f,"Color",new Variant(0x000000FF));
		Dispatch.put(f,"Italic",new Variant(true));
		Dispatch.put(f,"Angle",new Variant(5));	

		//j.DrawImage(40,10,32,32,StartupDir()+"..\\Data\\WSO.ICO")
	
		t = Dispatch.call(j,"TextOut",new Variant(100),new Variant(50),new Variant(Text)).toDispatch();
		f = Dispatch.get(t,"Font").toDispatch();
		Dispatch.put(f,"Bold",new Variant(true));

		Dispatch.call(j,"TextOut",new Variant(100),new Variant(90),new Variant("(C) Veretennikov A. B. 2004 - 2006"));			

		t = Dispatch.call(j,"CreateHyperLink",new Variant(100),new Variant(130),new Variant(200),
		new Variant(25),Dispatch.get(Version,"Url"),new Variant("WSO in the Web")).toDispatch();
		Dispatch.put(t,"Color",new Variant(0x00FFEEEE));

		boolean Trial = Dispatch.get(Version,"Trial").toBoolean();
		if (Trial)
			Dispatch.call(j,"TextOut",new Variant(100),new Variant(170),new Variant("It is a Demo version of WSO\rwith several serious limitations.\rTry it and buy the full version."));

		Dispatch.put(j,"Text",new Variant("About WSO"));
		new DispatchEvents(j,new FormEvents());

		Dispatch.call(j,"ShowModal");
		
		Dispatch.call(j,"Destroy");
	};
	
	static class FormEvents
	{
		
		public void OnHitTest(Variant[] args)
		{
			Dispatch Sender = args[3].toDispatch();
			Dispatch.put(Sender,"Value",new Variant(HTCAPTION));
		};		
		
	};

	static class CloseItemEvents
	{
		
		public void OnClick(Variant[] args)
		{
			Dispatch Sender = args[0].toDispatch();
			Dispatch.call(Dispatch.get(Sender,"Form").toDispatch(),"Close");
		};
		
		public void OnExecute(Variant[] args)
		{
			OnClick(args);
		};
		
	};

}

File: Demo/Delphi/WSOTest.pas
//WindowSystemObject (WSO) example
//Copyright (C) Veretennikov A. B. 2004

unit WSOTest;

interface

procedure AboutWSO(Sender:OleVariant; const s:String);

implementation

uses
  Windows,
  SysUtils,
  WSO,
  WSOConsts,
  ShellApi;

procedure CloseFormHandler(Sender:OleVariant);
begin
	Sender.Form.Close;
end;

procedure OnHitTest(Sender,x,y,ResultPtr:OleVariant);
begin
	ResultPtr.put(HTCAPTION);
end;

function StartupDir:String;
begin
  Result := ExtractFilePath(ParamStr(0));
end;

procedure WSOHelp(Sender:OleVariant);
begin
  ShellExecute(0,nil,PChar(StartupDir() + '..\..\..\wso.chm'),'','',SW_SHOW);
end;

procedure AboutWSO(Sender:OleVariant; const s:String);
Var
	j,t:OleVariant;
begin
	j := Sender.Form.CreateDialogForm(0,0,0,0);
	j.ClientWidth := 400;
	j.ClientHeight := 250;
	j.CenterControl;
	t := j.Rectangle(0,0,0,75);
  t.Brush.Color := $00FFA0A0;
  t.Brush.Style := B_GRADIENT or DIRECTION_TOP_BOTTOM;
  t.Brush.GradientColor := $00FFEEEE;
  t.Pen.Color := $00FF5555;
	t.Align := AL_TOP;

	t := j.Rectangle(0,0,0,0);
  t.Brush.Color := $00FFEEEE;
  t.Brush.Style := B_GRADIENT or DIRECTION_TOP_BOTTOM;
  t.Brush.GradientColor := $00FFEEEE;
  t.Pen.Color := $00FF5555;
  t.Align := AL_CLIENT;

	t := j.CreateButton(220,220,75,25,'Close');
	t.OnClick := TEventHandler.Create(CloseFormHandler);
	t.Default := true;
	t.Cancel := true;
	t := j.CreateButton(300,220,75,25,'Help');
	t.OnClick := TEventHandler.Create(WSOHelp);
	j.HelpButton := t;
	t := j.TextOut(100,10,j.Root.Version.String);
	t := t.Font;
	t.Bold := true;
	t.Name := 'Arial Black';
	t.Size := 14;
	t.Color := $000000FF;
	t.Bold := true;
	t.Italic := true;
	t.Angle := 5;
	j.DrawImage(40,10,32,32,StartupDir()+'..\..\Data\WSO.ICO');
	j.TextOut(100,50,s).Font.Bold := true;
	j.TextOut(100,90,'(C) Veretennikov A. B. 2004 - 2006');
	j.CreateHyperLink(100,130,200,25,j.Root.Version.Url,'WSO in the Web');
	j.Color := $00FFEEEE;
	if j.Root.Version.Trial then
	j.TextOut(100,170,'It is a Demo version of WSO'#13'with several serious limitations.'#13'Try it and buy the full version.');
	j.Text := 'About WSO';
	j.OnHitTest := TEventHandler.Create(OnHitTest);
	j.ShowModal;
	j.Destroy;
end;

end.
